src/shared/api/client.ts

```diff
@@ -0,0 +1,7 @@
+const API_BASE = import.meta.env.VITE_API_BASE_URL;
+
+export async function api<T>(path: string, init?: RequestInit): Promise<T> {
+  const res = await fetch(`${API_BASE}${path}`, init);
+  if (!res.ok) throw new Error(`HTTP ${res.status}`);
+  return res.json() as Promise<T>;
+}
```

Comment:
지훈님 말씀처럼 약간 이런식으로 api를 개선하면 api.get / api.post / api.put 같은 함수를 써볼 수 있을 것 같습니다. (그대로 쓰면 타입 오류날 수도 있어요)

```suggestion
export const api = {
  get: async <T>(path: string) => {
    const res = await fetch(`${API_BASE}${path}`, {
      method: "GET",
      headers: { "Content-Type": "application/json" },
    });
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json() as Promise<T>;
  },
  post: async <T, Body extends object>(path: string, body: Body) => {
    const res = await fetch(`${API_BASE}${path}`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(body),
    });
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json() as Promise<T>;
  },
  put: async <T, Body extends object>(path: string, body: Body) => {
    const res = await fetch(`${API_BASE}${path}`, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(body),
    });
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json() as Promise<T>;
  },
  ...
}
```